scan/base.go

```diff
@@ -40,6 +42,303 @@ type base struct {
+	Title         string     `json:"title"`
+	CreatedAt     string     `json:"created_at"`
+	UpdatedAt     string     `json:"updated_at"`
+	PublishedDate string     `json:"Published_date"`
+	VulnType      string     `json:"Vuln_type"`
+	References    References `json:"references"`
+	FixedIn       string     `json:"fixed_in"`
+}
+
+//References is for wpvulndb's json
+type References struct {
+	URL     []string `json:"url"`
+	Cve     []string `json:"cve"`
+	Secunia []string `json:"secunia"`
+}
+
+func detectWp(c *base) (vinfos []models.VulnInfo, err error) {
+
+	var coreVuln []models.VulnInfo
+	if coreVuln, err = detectWpCore(c); err != nil {
+		return
+	}
+	for _, i := range coreVuln {
+		vinfos = append(vinfos, i)
+	}
+
+	var themeVuln []models.VulnInfo
+	if themeVuln, err = detectWpTheme(c); err != nil {
+		return
+	}
+	for _, i := range themeVuln {
+		vinfos = append(vinfos, i)
+	}
+
+	var pluginVuln []models.VulnInfo
+	if pluginVuln, err = detectWpPlugin(c); err != nil {
+		return
+	}
+	for _, i := range pluginVuln {
+		vinfos = append(vinfos, i)
+	}
+
+	return
+}
+
+func detectWpCore(c *base) (vinfos []models.VulnInfo, err error) {
+	cmd := fmt.Sprintf("wp core version --path=%s", c.ServerInfo.WpPath)
+
+	var coreVersion string
+	if r := exec(c.ServerInfo, cmd, noSudo); r.isSuccess() {
+		tmp := strings.Split(r.Stdout, ".")
+		coreVersion = strings.Join(tmp, "")
+		coreVersion = strings.TrimRight(coreVersion, "\r\n")
+		if len(coreVersion) == 0 {
+			return
+		}
+	}
+	cmd = fmt.Sprintf("curl -k -H 'Authorization: Token token=%s' https://wpvulndb.com/api/v3/wordpresses/%s", c.ServerInfo.WpToken, coreVersion)
+	if r := exec(c.ServerInfo, cmd, noSudo); r.isSuccess() {
+		coreConvertVinfo(r.Stdout)
+	}
+	return
+}
+
+func coreConvertVinfo(stdout string) (vinfos []models.VulnInfo, err error) {
+	data := map[string]WpCveInfos{}
+	if err = json.Unmarshal([]byte(stdout), &data); err != nil {
+		return
+	}
+	for _, i := range data {
+		if len(i.Vulnerabilities) == 0 {
+			continue
+		}
+		for _, e := range i.Vulnerabilities {
+			if len(e.References.Cve) == 0 {
+				continue
+			}
+			NotFixedYet := false
+			if len(e.FixedIn) == 0 {
+				NotFixedYet = true
+			}
+			var cveIDs []string
+			for _, k := range e.References.Cve {
+				cveIDs = append(cveIDs, "CVE-"+k)
+			}
+
+			for _, cveID := range cveIDs {
+				vinfos = append(vinfos, models.VulnInfo{
+					CveID: cveID,
+					CveContents: models.NewCveContents(
+						models.CveContent{
+							CveID: cveID,
+							Title: e.Title,
+						},
+					),
+					AffectedPackages: models.PackageStatuses{
+						{
+							NotFixedYet: NotFixedYet,
+						},
+					},
+				})
+			}
+		}
+	}
+	return
+}
+
+//WpStatus is for wp command
+type WpStatus struct {
+	Name    string `json:"-"`
+	Status  string `json:"-"`
+	Update  string `json:"-"`
+	Version string `json:"-"`
+}
+
+func detectWpTheme(c *base) (vinfos []models.VulnInfo, err error) {
+	cmd := fmt.Sprintf("wp theme list --path=%s", c.ServerInfo.WpPath)
```

Comment:
Use `--format=json` option in order to easy parse.
It's difficult to handle Line feed `\n` when the term.

see 
https://github.com/future-architect/vuls/issues/689
https://github.com/future-architect/vuls/search?q=width&type=Commits

Reply:
implement